keps/sig-release/2853-kubernetes-kubernetes-repository-branch-rename/README.md

```diff
@@ -0,0 +1,583 @@
+
+Our biggest repository is [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes), and this has a lot of open Pull Requests and several other forks, clones that use that for downstream signal or a custom build.
+
+Changing the branch name might cause issues for downstream consumers, our contributors, and the community in general.
+
+For that, we are creating this KEP to have a plan to address the issues and communication.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+The main goal is to be more inclusive using better naming.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+This applies only to the main Kubernetes GitHub repository [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes).
+Other repositories will follow a similar approach but at their timeframe.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+- We aim to make the change during the start of v1.25 release (spring 2022).
+- Perform a Survey to gather information on downstream consumers and how this might affects their workflow. See [Survey Questions](#survey-questions).
+- Announce the change in the kubernetes-dev and kubernetes-announcement mailing list and in a blog post. As well as use Twitter and other social media to announce.
+- Change all open Pull Requests to `Draft mode` (so that when we rename the branch, tests will not be triggered, and we avoid a massive queue in the CI infrastructure).
+- Update all Prow jobs that use `kubernetes/kubernetes` and use the `master` branch to listen to the `main` branch as well
+- Follow the guide on https://github.com/kubernetes/community/blob/master/github-management/default-branch-migration.md to rename the repository branch name.
+- After the branch rename, announce that in the mailing list
+- Update all Prow jobs that use `kubernetes/kubernetes` and to remove the `master` branch
+- Convert on demand the open Pull Requests from the `Draft mode` to `Review ready mode`
+
+
+References:
+- https://github.com/kubernetes/org/issues/2222
+- https://github.com/kubernetes/enhancements/issues/2853
+- https://github.com/kubernetes/kubernetes/issues/105601
+- https://github.com/kubernetes/community/blob/master/github-management/default-branch-migration.md
+
+- https://github.com/github/renaming
+
+### Risks and Mitigations
```

Comment:
Have we done any research on how is it going to affect us (speaking of Release Engineering)? For example:

* What changes we might need to do to krel and the release tooling?
* What about the release notes tool?
* What about the version markers (could they be affected by this)?

Reply:
> What changes we might need to do to krel and the release tooling?
> What about the release notes tool?

we will need to update some default const we have to use main, or do some fallback if fail, like try master and if fail go to main, or otherwise

> What about the version markers (could they be affected by this)?

regarding this maybe @saschagrunert or @justaugustus can give some opinion, I'm not expert in this topic :(

Reply:
Any tooling would require to be adapted. So there is a risk we miss some of them, which can be mitigated by grepping with multiple eyes :)

Reply:
will add this as a risk and mitigation plans